### PR TITLE
Final Group Capstone - Rent a Car - Back-End

### PR DESCRIPTION
**Pull request by ezraermy**
#### In this pull request, @yosaddis, @PabloEGonz and I implemented the following features:
-  Created user authentication for login and registration.
-  Created necessary API endpoints for Users, Cars, and Reservations.
-  Created a CRUD functionality for Cars
-  Created a CRUD functionality for Reservations

#### Additional Features 
-  Extended the API to support car deletion, including the ability to mark cars as removed from the database  
- Implemented API endpoints to retrieve and display user-specific booking form to rent a new car
-  Implemented API endpoints to retrieve and display user-specific booking information for list of rents.
- Followed gitflow.
- worked in a professional way.
- Fix linter errors.
  - Create a unit test for each model.
Screenshot from 2023-10-02 23-56-25

#### A link to our front-end pull request:
- [full-stack-capstone-front-end](https://github.com/PabloEGonz/full-stack-capstone-front-end/pull/23)

#### Important :red_circle: 
- Kindly open the [render-deployed back-end](https://rent-a-car-cfdo.onrender.com/api/cars) link before opening the [front-end live demo](https://rent-a-car-front-end.onrender.com/) due to the cold start in the back end deployed in render.
- If you get an JSON error when you open the front-end page, go to the local storage and delete the key.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,167 @@
...
+
+### Prerequisites
+
+In order to run this project you need to:
+
+- Install [Git](https://git-scm.com/)
+- Install [Ruby](https://www.ruby-lang.org/en/)
+- Install a code editor, I suggest [VsCode](https://code.visualstudio.com/)
+
+### Setup
```
Comment (Meltrust) [inline README.md:67]: - [ ] Kindly, include in your instructions the required db commands that your users will need to run your backend app: ``rails db:create, rails db:migrate``. 👍

File: README.md [hunk 2]
```diff
@@ -1,2 +1,167 @@
...
+
+Install the dependencies using npm:
+
+```bash
+bundler install
+```
+
+
+
+### Run tests
```
Comment (Meltrust) [inline README.md:95]: - [ ] You have some functional model tests, but there are no instructions about how to run them.

Kindly, include a "tests" section on your readme with specific instructions about how to run your models tests:

image

File: README.md [hunk 3]
```diff
@@ -1,2 +1,167 @@
-# Full-Stack-Capstone-Backend
-Explore the backend of Doctor Reservation App! This repository, powered by Rails, manages user authentication, reservation logic, and API endpoints.
+# Full Stack Capstone Backend<a name="readme-top"></a>
```
Comment (Meltrust) [inline README.md:1]: - [ ]  You are using Rspec, but the "test" folder still exists, which is used by another test framework. Kindly consider deleting that folder if you are not using it. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @ezraermy,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Model tests are passing✔️
-  Good readme ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Review by Meltrust — APPROVED** [review]
#### Hi @ezraermy,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Readme has been improved✔️ 
- Superfluous test folder has been removed✔️ 
- Great job✔️ 



Cheers and Happy coding!👏👏👏


------

**Comment by ezraermy** [comment]
Hi @Meltrust. We thank you very much for your tips and feedback that surely imporves our project features and functionality.

**Pull request merged**